### PR TITLE
planner: add some tiflash test cases on dynamic partition prune mode

### PR DESCRIPTION
**Pull request by rebelice**
### What problem does this PR solve?

Issue Number: related to #24150 

Problem Summary: planner: add some test cases about tiflash on dynamic mode

### What is changed and how it works?

planner: add more test cases about tiflash and dynamic mode

### Check List 

Tests 

- Unit test

### Release note 

- planner: add some test cases about tiflash on dynamic mode

### CODE REVIEW

File: executor/tiflash_test.go [hunk 1]
```diff
@@ -526,3 +528,142 @@ func (s *tiflashTestSuite) TestTiFlashVirtualColumn(c *C) {
...
+	vals := make([]string, 0, 100)
+	for i := 0; i < 100; i++ {
+		vals = append(vals, fmt.Sprintf("(%v, %v)", rand.Intn(400), rand.Intn(400)))
+	}
+	for _, tbl := range []string{`thash`, `trange`, `tlist`, `tnormal`} {
+		tk.MustExec(fmt.Sprintf("insert into %v values %v", tbl, strings.Join(vals, ", ")))
+		tk.MustExec(fmt.Sprintf("analyze table %v", tbl))
+	}
+	tk.MustExec("set @@session.tidb_isolation_read_engines='tiflash'")
+	tk.MustExec("set @@session.tidb_allow_mpp=ON")
```
Comment (qw4990) [inline executor/tiflash_test.go:573]: Please set it to `2` to enforce TiDB to use MPP.

**Review by hanfei1991 — APPROVED** [review]

**Review by qw4990 — APPROVED** [review]

**Pull request merged**